src/items/functions.md

```diff
@@ -56,7 +56,7 @@ r[items.fn.signature]
 Functions may declare a set of *input* [*variables*][variables] as parameters, through which the caller passes arguments into the function, and the *output* [*type*][type] of the value the function will return to its caller on completion.
 
 r[items.fn.implicit-return]
-If the output type is not explicitly stated, it is the [unit type].
+If the output type is not explicitly stated, it is the [unit type]. However, if the block expression is not [diverging](../divergence.md), then the output type is instead [`!`](../types/never.md).
```

Comment:
In talking over this PR in the office hours, @ehuss and I puzzled over this rule.  Presumably the *not* is misplaced, but even with that removed, we weren't clear on how to assign meaning to this given:

```rust
fn f() {
    loop {} // Diverging expression.
}

let _ = || -> ! { f() }; //~ ERROR mismatched types
let _: fn() -> ! = f; //~ ERROR mismatched types
```